### PR TITLE
chore: Update default rust toolchain to 1.75

### LINKED ISSUE #310: Make rust version a single workflow variable
It would be better to change the version in one place, instead across many steps.

Example (stripped for brevity and clarity):

```yml
env:
  RUST_TOOLCHAIN_VERSION: "1.75.0"

jobs:
  run_clippy:
    steps:
      - uses: dtolnay/rust-toolchain@master
        with:
          toolchain: ${{ env.RUST_TOOLCHAIN_VERSION }}
          components: clippy
```

`operator-rs` already does so:
- https://github.com/stackabletech/operator-rs/blob/407cfec12469dba32ac1795a9ce7e62077a0650b/.github/workflows/build.yml#L20
- https://github.com/stackabletech/operator-rs/blob/407cfec12469dba32ac1795a9ce7e62077a0650b/.github/workflows/build.yml#L86-L89

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
Upgrade the default rust toolchain version to 1.75

Here is a test run against airflow-operator:

~~~patch
diff --git a/nix/sources.json b/nix/sources.json
index 92a5322..ca7ce78 100644
--- a/nix/sources.json
+++ b/nix/sources.json
@@ -17,10 +17,10 @@
         "homepage": "",
         "owner": "NixOS",
         "repo": "nixpkgs",
-        "rev": "3f21a22b5aafefa1845dec6f4a378a8f53d8681c",
-        "sha256": "15y8k3hazg91kscbmn7dy6m0q6zvmhlvvhg97gcl5kw87y0svzxk",
+        "rev": "5ad9903c16126a7d949101687af0aa589b1d7d3d",
+        "sha256": "1i0nvgzzadbl29hzs5n4qbc0nnw69nh79b0kq3g7zi1926rczlqn",
         "type": "tarball",
-        "url": "https://github.com/NixOS/nixpkgs/archive/3f21a22b5aafefa1845dec6f4a378a8f53d8681c.tar.gz",
+        "url": "https://github.com/NixOS/nixpkgs/archive/5ad9903c16126a7d949101687af0aa589b1d7d3d.tar.gz",
         "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
     }
 }
diff --git a/rust-toolchain.toml b/rust-toolchain.toml
index 639f4f1..7897a24 100644
--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.74.0"
+channel = "1.75.0"

~~~

### CODE REVIEW

**Comment by sbernauer** [comment]
Can you please also make sure `make run-dev` uses Rust 1.75.0? I had real trouble to get trino-lb running using nix and Rust 1.75.0

**Comment by NickLarsenNZ** [comment]
> Actual change LGTM. Can you please also make sure `make run-dev` uses Rust 1.75.0? I had real trouble to get trino-lb running using nix and Rust 1.75.0

@sbernauer do you mean test an operator that has been upgraded to 1.75.0?
If so, `make run-dev` works for the [upgraded hdfs-operator](https://github.com/stackabletech/hdfs-operator/pull/454/files#diff-2b1bde2cf3a858b7bf7424cb8bcbf01f35b94dc80b925d9432cbab3319ca9b4e).

Let me know if there was something to look out for in particular?

**Comment by sbernauer** [comment]
Could you please add the following snippet to the code and test if `make run-dev` still works?
```
trait Foo {
    async fn foo();
}
```

**Comment by fhennig** [comment]
We could also use an organization-level variable that is shared across all repos to set the toolchain version. I though we already had that, but it doesn't seem like it actually. What do you think?

**Comment by NickLarsenNZ** [comment]
> We could also use an organization-level variable that is shared across all repos to set the toolchain version. I though we already had that, but it doesn't seem like it actually. What do you think?
> 
> Github docs: https://docs.github.com/en/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows

@fhennig, I like the central nature of that but not removing it from source control. Github Actions Secrets and Variables are useful for data that isn't useful or allowed to be tracked in a repository. Just my thoughts on that.

**Comment by NickLarsenNZ** [comment]
> Could you please add the following snippet to the code and test if `make run-dev` still works?
> 
> ```
> trait Foo {
>     async fn foo();
> }
> ```
> 
> Because if fear tilt might still build with rustc 1.74.0

I tested this and it's fine as long as nixpkgs is up to date, eg: `niv update nixpkgs`.

**Review by Techassi — DISMISSED** [review]

File: repositories.yaml [hunk 1]
```diff
@@ -82,7 +82,7 @@ repositories:
 # IMPORTANT
 # If you change the Rust toolchain version here, make sure to also change
 # Docker-images/ubi8-rust-builder/Dockerfile
-rust_version: 1.74.0
+rust_version: 1.75.0
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply repositories.yaml:85]: There's about to be a docker-images PR ;)

[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply repositories.yaml:85]: https://github.com/stackabletech/docker-images/pull/542

**Comment by sbernauer** [comment]
> I tested this and it's fine as long as nixpkgs is up to date, eg: niv update nixpkgs.

That's exactly my point ;) Please update https://github.com/stackabletech/operator-templating/blob/main/template/nix/sources.json as well (preferable only nixpkgs and *not* crate2nix)

**Comment by NickLarsenNZ** [comment]
> > I tested this and it's fine as long as nixpkgs is up to date, eg: niv update nixpkgs.
> 
> That's exactly my point ;) Please update https://github.com/stackabletech/operator-templating/blob/main/template/nix/sources.json as well (preferable only nixpkgs and _not_ crate2nix)

@sbernauer oh right. I somehow assumed that was all done in the operators themselves.

Would you prefer I make a separate PR for the version bump? The main purpose of this change was to replace the many version strings with a variable in CI. That allows operators to be bumped independently if needed or to test the waters.

**Comment by NickLarsenNZ** [comment]
@sbernauer I've updated the nixpkgs. But happy to pull the version changes out into a separate PR.

**Comment by NickLarsenNZ** [comment]
I'm going to mover the CI variable changes into a new PR so it can be merged independently of Rust version bumping.

**Review by Techassi — APPROVED** [review]

**Review by sbernauer — APPROVED** [review]

**Pull request merged**